src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -19,15 +19,27 @@
 class DeprecationErrorHandler
 {
     const MODE_WEAK = 'weak';
+    const MODE_WEAK_VENDORS = 'weak_vendors';
     const MODE_DISABLED = 'disabled';
 
     private static $isRegistered = false;
 
     /**
+     * @var string[] absolute paths to vendor directories
+     */
+    private static $vendors;
+
+    /**
+     * For test purposes. Do not use.
+     */
+    public static $testMode = false;
```

Comment:
We should rather write a test creating a fake vendor folder with its autoloading file (similar to the one generated by Composer).

Having this argument here means that your test will **not** ensure that the feature is working (try to replace the normal logic with just ``return false`` and see you tests passing)